tests/integration/test_dynamic_sampling.py

```diff
@@ -239,11 +250,38 @@ def test_it_removes_events(mini_sentry, relay, rule_type, event_factory):
     assert outcome.get("reason") == f"Sampled:{rules[0]['id']}"
 
 
-@pytest.mark.parametrize(
-    "rule_type, event_factory",
-    [("error", _create_event_envelope), ("transaction", _create_transaction_envelope)],
-)
-def test_it_keeps_events(mini_sentry, relay, rule_type, event_factory):
+def test_it_does_not_sample_error(mini_sentry, relay):
+    """
+    Tests that we keep an event if it is of type error.
+    """
+    project_id = 42
+    relay = relay(mini_sentry, _outcomes_enabled_config())
+
+    # create a basic project config
+    config = mini_sentry.add_basic_project_config(project_id)
+    public_key = config["publicKeys"][0]["publicKey"]
+
+    # add a sampling rule to project config that removes all transactions (sample_rate=0)
+    rules = _add_sampling_config(config, sample_rate=0, rule_type="transaction")
```

Comment:
Let's make this a trace rule and also add a DSC to the envelope to ensure we're testing the right thing.